### PR TITLE
adding basic behaviors

### PR DESCRIPTION
**Pull request by rohanbayya**
breaking up [this](https://github.com/Instawork/hyperview/pull/895/files) pr

### CODE REVIEW

File: examples/behaviors/basic/actions/append/once/style.xml.njk [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<style
```
Comment (flochtililoch) [inline examples/behaviors/basic/actions/append/once/style.xml.njk:1]: is this needed? the button style should already be defined in the macro right?

File: examples/behaviors/basic/actions/prepend/prepend-once/style.xml.njk [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<style
+  id="button"
```
Comment (flochtililoch) [inline examples/behaviors/basic/actions/prepend/prepend-once/style.xml.njk:2]: same question here

File: examples/behaviors/basic/actions/replace-inner/nested/_replace_fragment_inner2.xml.njk [hunk 3]
```diff
@@ -0,0 +1,3 @@
+{% from 'macros/button/index.xml.njk' import button %}
+
+{{ button('Press me again...', {"action": "replace-inner", "href":"/behaviors/basic/actions/replace-inner/nested/_replace_fragment_inner2.xml", "xmlns":"https://hyperview.org/hyperview"}) }}
```
Comment (flochtililoch) [inline examples/behaviors/basic/actions/replace-inner/nested/_replace_fragment_inner2.xml.njk:3]: I think we should make the behavior an explicit XML element here - separating the UI from the "content"

File: examples/behaviors/basic/actions/replace-inner/nested/index.xml.njk [hunk 4]
```diff
@@ -0,0 +1,5 @@
+{% from 'macros/button/index.xml.njk' import button %}
+{% from 'macros/description/index.xml.njk' import description %}
+
+{{ description('In this example, the replaced element also has a replace-inner action, allowing for a nested interaction.') }}
+{{ button('Press me', {action: "replace-inner", href:"/behaviors/basic/actions/replace-inner/nested/_replace_fragment_inner2.xml"}) }}
```
Comment (flochtililoch) [inline examples/behaviors/basic/actions/replace-inner/nested/index.xml.njk:5]: same here

File: examples/behaviors/basic/actions/replace-inner/once/index.xml.njk [hunk 5]
```diff
@@ -0,0 +1,6 @@
+{% from 'macros/button/index.xml.njk' import button %}
+{% from 'macros/description/index.xml.njk' import description %}
+
+{{ description('Tapping the button below will request an XML fragment and replace the children of the element with the response.') }}
+{# behavior attributes can be attached to a view too #}
+{{ button('Press me', attributes={"action": "replace-inner", "href":"/behaviors/basic/actions/replace-inner/once/_replace_fragment_inner.xml"}) }}
```
Comment (flochtililoch) [inline examples/behaviors/basic/actions/replace-inner/once/index.xml.njk:6]: same here

File: examples/behaviors/basic/actions/replace/once/index.xml.njk [hunk 6]
```diff
@@ -0,0 +1,6 @@
+{% from 'macros/button/index.xml.njk' import button %}
+{% from 'macros/description/index.xml.njk' import description %}
+
+{{ description('Tapping the button below will request an XML fragment and replace the element with the response.') }}
+{# behavior attributes can be attached to a view too #}
+{{ button('Press me', attributes={"action": "replace", "href":"/behaviors/basic/actions/replace/once/_replace_fragment.xml"}) }}
```
Comment (flochtililoch) [inline examples/behaviors/basic/actions/replace/once/index.xml.njk:6]: same here. Applies to every other behavior examples where the raw XML behavior markup is considered to be the "content" and should be surfaced in the file directly, as opposed to the button itself that is the part of the "UI" and ok to abstract out.

File: examples/behaviors/basic/actions/unselect-all/indicator/index.xml.njk [hunk 7]
```diff
@@ -0,0 +1,42 @@
+{% from 'macros/button/index.xml.njk' import button %}
+{% from 'macros/description/index.xml.njk' import description %}
+
+{{ description('When selecting with delay, you can use indicators.') }}
+
+<select-multiple id="id_sm_indicator" name="sm" style="select">
```
Comment (flochtililoch) [inline examples/behaviors/basic/actions/unselect-all/indicator/index.xml.njk:6]: I think there's an opportunity to reuse this common markup (and associated styles) across all examples of select-multiple behaviors

File: examples/_includes/templates/section-list.xml.njk [hunk 8]
```diff
@@ -33,7 +33,7 @@ hv_section_list_tag: "root"
           <text style="item-label">{{ screen.data.hv_title }}</text>
         </view>
         <view style="item-chevron">
-          {% include './icons/chevron-right.svg' %}
+          {% include 'icons/chevron-right.svg' %}
```
Comment (rohanbayya) [inline examples/_includes/templates/section-list.xml.njk:36]: test on web view

File: examples/behaviors/basic/actions/append/once/style.xml.njk (hunk 1 above)
Comment (rohanbayya) [reply examples/behaviors/basic/actions/append/once/style.xml.njk:1]: i'm tweaking the button style here. flexdirection = column

File: examples/behaviors/basic/actions/prepend/prepend-once/style.xml.njk (hunk 2 above)
Comment (rohanbayya) [reply examples/behaviors/basic/actions/prepend/prepend-once/style.xml.njk:2]: tweaking the flexDirection

File: examples/behaviors/basic/actions/replace-inner/nested/_replace_fragment_inner2.xml.njk (hunk 3 above)
Comment (rohanbayya) [reply examples/behaviors/basic/actions/replace-inner/nested/_replace_fragment_inner2.xml.njk:3]: I added this to demonstrate the idea that button can have the behavior attributes as opposed to passing a <behavior/> explicitly. I will revert this if seems unecessary to the user

File: examples/behaviors/basic/actions/unselect-all/indicator/index.xml.njk (hunk 7 above)
Comment (rohanbayya) [reply examples/behaviors/basic/actions/unselect-all/indicator/index.xml.njk:6]: yes, all the examples use common styles defined in https://github.com/Instawork/hyperview/pull/904/files#diff-f510acce2d03d8e603547ee73aec6e6802a4da558929545f954f63f27711a22f.
added a macro for re-using markup

File: examples/behaviors/basic/actions/append/index.xml.njk [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+
+{% block styles %}
+  {% include './once/style.xml.njk' %}
+  {% include './nested/style.xml.njk' %}
+{% endblock %}
+
+{% block content %}
+  {% include './once/index.xml.njk' %}
+  {% include './nested/index.xml.njk' %}
+{% endblock %}
```
[the remark this answers is not in the document]
Comment (rohanbayya) [reply examples/behaviors/basic/actions/append/index.xml.njk:17]: clarified that this doesnot seem like an overkill

File: examples/behaviors/basic/actions/replace-inner/nested/_replace_fragment_inner2.xml.njk (hunk 3 above)
Comment (rohanbayya) [reply examples/behaviors/basic/actions/replace-inner/nested/_replace_fragment_inner2.xml.njk]: passing <behavior/> as per our discussion

File: examples/_includes/templates/section-list.xml.njk (hunk 8 above)
Comment (rohanbayya) [reply examples/_includes/templates/section-list.xml.njk:36]: tested, working fine

File: examples/behaviors/basic/actions/append/once/style.xml.njk (hunk 1 above)
Comment (flochtililoch) [reply examples/behaviors/basic/actions/append/once/style.xml.njk:1]: I think this property should probably be part of the root button style anyway?

**Review by flochtililoch — APPROVED** [review]

**Pull request merged**